pkg/operator/update/resources.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2020 PlanetScale Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package update
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	planetscalev2 "planetscale.dev/vitess-operator/pkg/apis/planetscale/v2"
+)
+
+func KeyspaceDiskSize(dst *planetscalev2.VitessKeyspaceTemplate, src planetscalev2.VitessKeyspaceTemplate) {
+	// Check that the keyspace definitions line up.
+	if !validateKeyspacePartitionings(*dst,src) {
+		return
+	}
+
+	updateDiskSize(dst, src)
+}
+
+func updateDiskSize(dst *planetscalev2.VitessKeyspaceTemplate, src planetscalev2.VitessKeyspaceTemplate) {
+	for i := range dst.Partitionings {
+		dstPartitioning := dst.Partitionings[i]
```

Comment:
The purpose of the index trick is so we can make this a pointer instead of a copy:

```suggestion
		dstPartitioning := &dst.Partitionings[i]
```